### PR TITLE
Add TAnon comparison to fast_eq

### LINKED ISSUE #8588: Compiler hanging on invalid `@:from`
```haxe
typedef Abstract = Under<{}>;

@:multiType abstract Under<T>(T) {

    function new(v:T);

    @:from public static inline function from<T>(v:T) { return /*cast*/ v; } // { return new Under(v); }
//    @:to public inline function to(v:T) { return v; }
}

        var a1:Abstract = { value:(null:Dynamic), type:TUnknown };
```

Casting return of @:from function or using new-@:to combination works without hanging

### PR DESCRIPTION
**Pull request by RealyUniqueName**
Fixes #8588 
I'm not sure this is a correct fix. And `fast_eq` doesn't look so fast now.
If this is accepted then probably `fast_eq_mono` should be updated too.

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
Moved that implementation to a new `fast_eq_anon`

**Comment by Simn** [comment]
Please don't merge this yet, I want to check if `get_abstract_froms` is really doing the right thing first.

File: src/core/type.ml [hunk 1]
```diff
@@ -1729,42 +1729,74 @@ let link_dynamic a b = match follow a,follow b with
...
+						let f1 = PMap.find f1 a1.a_fields
+						and f2 = PMap.find f2 a2.a_fields in
+						fast_eq_anon f1.cf_type f2.cf_type
+					with Not_found -> false)
+					&& loop rest1 rest2
+			in
+			loop (List.sort compare !fields1) (List.sort compare !fields2)
+		in
+		(match !(a2.a_status), !(a1.a_status) with
+		| Statics c, Statics c2 when c == c2 -> fields_eq()
```
Comment (Simn) [inline src/core/type.ml]: Do we really need the `fields_eq()` here?

**Comment by Simn** [comment]
So `get_abstract_froms` is definitely a bit weird. At the very least it's missing a `follow t`, which would trigger some checks already and detect `@:from` fields which are outright invalid. However, that doesn't deal with the more general situation, so I think something like this PR is actually necessary.

However, why are we only using this in one place? There are two more occurrences in typer.ml which should suffer from the same problems.

File: src/core/type.ml (hunk 1 above)
Comment (RealyUniqueName) [reply src/core/type.ml]: Nope, we don't

**Pull request merged**